Adds WatchKit support via CocoaPods

leoAsana opened the pull request:
As mentioned in my issue ##464 was was looking to use the tracker from the Apple Watch directly.

I went ahead and fixed the issues that prevented the project to build on watchOS and added watchOS to the podspec. 

The changes a just some more `#if`s in `SPUtilities.m` and I had to replace the use of `NSURLConnection` in `SPEmitter.m` with `NSURLSession`. I used a `dispatch_semaphore` to make it synchronous as mentioned [here](https://forums.developer.apple.com/thread/11519).

I've also added an watch app to the Swift Demo and it worked in my tests.

snowplowcla commented:
Thanks for your pull request. Is this your first contribution to a Snowplow open source project? Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).

:memo: Please visit https://github.com/snowplow/snowplow/wiki/CLA to learn more and sign.

Once you've signed, please reply here (e.g. I signed it!) and we'll verify. Thanks.

leoAsana commented:
I signed it!

AlexBenny commented:
Thanks for your contribute @leoAsana .
As you noticed some parts of the tracker haven't been touched for a while. That's the case of the WatchOS target, so your contribute is very valuable.

We are going to release soon the version 1.1.4 (early next week). Please, could you rebase your branch onto master as soon as we proceed with the release of 1.1.4 ?

leoAsana commented:
I've rebased `master` and edited the new example projects for `carthage` and `cocoapods` to include a watch target.
I also fixed some build settings problems on the watch target, so now it works with `Carthage` to.

I know this is a rather large PR, but most changes are just in the example projects.

AlexBenny commented:
Thanks @leoAsana. That's very helpful, I'll review the changes in the next days.
Meanwhile I've set the milestone to the next planned version 1.2.0

leoAsana commented:
@AlexBenny Thanks! I'm already using by fork in our watch project and I havn't seen any problems yet. I noticed that the platform is `pc`. Is that intended or should I change the value?

AlexBenny commented:
@leoAsana  Sorry for the delays, and happy to see that your solution is working without issues. Unfortunately we’re working on some of the core features, that slows down the integration of GitHub contributes, but once we have the high priority work done we will review your PR.

> I noticed that the platform is `pc`. Is that intended or should I change the value?

We haven't defined anything for wearables at the moment. I think it would make sense using `mob` for the WatchOS platform. Much of the watch apps are iOS apps extensions.

FYI: we just released the version 1.1.5 where we fixed an issue introduced on 1.1.4 about IDFA.

leoAsana commented on Snowplow/SPUtilities.m:
```diff
@@ -198,18 +205,26 @@ + (NSString *) getDeviceModel {
+    return [[UIDevice currentDevice] model];
+#elif SNOWPLOW_TARGET_WATCHOS
+    return [[WKInterfaceDevice currentDevice] model];
+#else
```
👍

leoAsana commented:
Thanks for the review. I applied all the request changes.

AlexBenny commented:
Thanks @leoAsana for your great work!
I've just merged your branch in our branch `464-add_watchkit_support` for rebasing with the last changes before to merge it to the release branch.
I'll keep you informed when done!
